### ISSUE TITLE
NOT WORK

### ISSUE THREAD

**Issue by xxxlee** [jisaacks/GitGutter#460]
GitGutter: disabled for "1.html" because file not in a working tree

GitGutter doesn't recognize working tree
Can you tell me the details in detail?
I do not understand what it is：<path_to_git_dir>.
![2017-10-14 4 57 42](https://user-images.githubusercontent.com/29938346/31574353-e63f656a-b093-11e7-9a53-e45b157c4c30.png)

**Comment by deathaxe** [comment]
GitGutter searches for either a `.git` file or directory in the open file's directory or one of its parents to determine the "root" of the repository. If it doesn't find such a `.git`, the file is considered not to be part of such a repository.

In a normal git repository, you'll always have a `.git` directory in the working-tree root, which holds the index and the tree (database) of your repository.

Beginning with git 2.5 or so it allows both, place the work-tree somewhere else and checkout multiple work-trees per database.

If you use `git worktree add` to checkout a second working tree of an already existing one, it will contain a `.git` file with the path to the real `.git` directory automatically. You can imagine it as a kind of link. This file is needed to let git know about the location of the database.

Example:

```
  /data
    |
    +- work_tree
    |  |
    |  +- .git        (the real git directory)
    |  |
    |  +- file        (the file opened in ST)
    |
    +- work_tree_copy
    |  |
    |  +- .git        a file containing "gitdir: /data/work_tree/.git."
    |  |
    |  +- file        maybe another version of "work_tree/file"
```

If your working tree does neither contain the real `.git` database folder nor such a link file, you'll need to create the link file manually in the root. Otherwise the folder is not recognized as working tree.

Just create a text file with

```
gitdir: /path/to/.git.
```

**Comment by xxxlee** [comment]
![2017-10-14 8 54 16](https://user-images.githubusercontent.com/29938346/31575842-ef91d56e-b0b4-11e7-8d37-b16f14253f25.png)
Why???

**Comment by xxxlee** [comment]
startup, version: 3143 osx x64 channel: stable
executable: /Applications/Sublime Text.app/Contents/MacOS/Sublime Text
working dir: /
packages path: /Users/lee/Library/Application Support/Sublime Text 3/Packages
state path: /Users/lee/Library/Application Support/Sublime Text 3/Local
zip path: /Applications/Sublime Text.app/Contents/MacOS/Packages
zip path: /Users/lee/Library/Application Support/Sublime Text 3/Installed Packages
ignored_packages: ["Vintage"]
Errors parsing theme:
icon_file_type is missing layer0.opacity, setting to 1.0 for backwards compatibility
pre session restore time: 0.683255
startup time: 0.731757
reloading plugin Default.auto_indent_tag
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.install_package_control
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.quick_panel
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.settings
reloading plugin Default.show_scope_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin Default.ui
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.01-pygments
reloading plugin 0_package_control_loader.50-backrefs
reloading plugin 0_package_control_loader.50-markupsafe
reloading plugin 0_package_control_loader.50-pymdownx
reloading plugin 0_package_control_loader.50-python-markdown
reloading plugin 0_package_control_loader.50-pyyaml
reloading plugin 0_package_control_loader.51-python-jinja2
reloading plugin 0_package_control_loader.55-mdpopups
reloading plugin A File Icon.A File Icon
reloading plugin AutoFileName.autofilename
reloading plugin AutoFileName.getimageinfo
reloading plugin BracketHighlighter.bh_core
reloading plugin BracketHighlighter.bh_logging
reloading plugin BracketHighlighter.bh_plugin
reloading plugin BracketHighlighter.bh_popup
reloading plugin BracketHighlighter.bh_regions
reloading plugin BracketHighlighter.bh_remove
reloading plugin BracketHighlighter.bh_rules
reloading plugin BracketHighlighter.bh_search
reloading plugin BracketHighlighter.bh_swapping
reloading plugin BracketHighlighter.bh_wrapping
reloading plugin BracketHighlighter.support
reloading plugin ChineseLocalizations.Localization
reloading plugin Git.git_commands
reloading plugin GitGutter.plugin
reloading plugin Material Theme.Icons
reloading plugin Material Theme.MT
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
reloading plugin SideBarEnhancements.SideBar
reloading plugin SideBarEnhancements.SideBarAPI
reloading plugin SideBarEnhancements.SideBarDefaultDisable
reloading plugin SublimeLinter-pep8.linter
SublimeLinter: pep8 linter loaded 
reloading plugin Anaconda.anaconda
reloading plugin Anaconda.setup
reloading plugin Anaconda.version
reloading plugin Python PEP8 Autoformat.pep8_autoformat
Included directory to sys.path : ['/Users/lee/Library/Application Support/Sublime Text 3/Packages/Python PEP8 Autoformat/libs', '/Users/lee/Library/Application Support/Sublime Text 3/Packages/Python PEP8 Autoformat/libs/py33']
reloading plugin SublimeLinter.commands
reloading plugin SublimeLinter.sublimelinter
reloading plugin SublimeREPL.__init__
reloading plugin SublimeREPL.completions
reloading plugin SublimeREPL.lang_integration
reloading plugin SublimeREPL.run_existing_command
reloading plugin SublimeREPL.sublimerepl
reloading plugin SublimeREPL.sublimerepl_build_system_hack
reloading plugin SublimeREPL.text_transfer
plugins loaded
SublimeLinter: debug mode: off 
environment variables loaded using: /bin/bash -l
Package Control: Skipping automatic upgrade, last run at 2017-10-14 21:22:05, next run at 2017-10-14 22:22:05 or after
SublimeLinter: pep8 activated: <builtin>

**Comment by deathaxe** [comment]
> Why???

You already answered:

> GitGutter: disabled for "1.html" because file not in a working tree

Does your folder "test" contain a directory called ".git"?

**Comment by xxxlee** [comment]
LeedeMacBook-Air:test lee$ git init
Initialized empty Git repository in /Users/lee/test/.git/
LeedeMacBook-Air:test lee$ ls -a
./                .git/             2.py
../               1.html            HelloWorld.class
.DS_Store         1.py              HelloWorld.java
LeedeMacBook-Air:test lee$ git status
On branch master

Initial commit

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	.DS_Store
	1.html
	1.py
	2.py
	HelloWorld.class
	HelloWorld.java

nothing added to commit but untracked files present (use "git add" to track)
LeedeMacBook-Air:test lee$ git config --add core.worktree /users/lee/test
LeedeMacBook-Air:test lee$ echo "gitdir: $(git rev-parse --git-dir)" > .git
-bash: .git: Is a directory
LeedeMacBook-Air:test lee$

**Comment by xxxlee** [comment]
LeedeMacBook-Air:.git lee$ git add .
LeedeMacBook-Air:.git lee$ echo "gitdir: $(git rev-parse --git-dir)" > .git
LeedeMacBook-Air:.git lee$ ls

**Comment by xxxlee** [comment]
![2017-10-14 10 41 32](https://user-images.githubusercontent.com/29938346/31576613-e6cc904a-b0c3-11e7-83ec-d12756e9e1ae.png)
????

**Comment by xxxlee** [comment]
vim .git
gitdir: /user/lee/test/.git         ????


git status
fatal: Not a git repository: /user/lee/test/.git

**Comment by deathaxe** [comment]
Not sure what you try to do, but guess it is too much ...

> LeedeMacBook-Air:test lee$ git config --add core.worktree /users/lee/test
LeedeMacBook-Air:test lee$ echo "gitdir: $(git rev-parse --git-dir)" > .git
-bash: .git: Is a directory
LeedeMacBook-Air:test lee$

Not needed at all.

If your working folder is `/users/lee/test` it is enough to call `git init` to turn the folder into a git repository. `git init` creates the _.git_ directory within _test_ directory, which GitGutter looks for.

Each file in _/users/lee/test_ is now untracked.

Calling `git add .` stages them. As they did not exist in git's database, they are marked as inserted (green + next to each line)

Calling `git commit` will remove the green +.

**Comment by xxxlee** [comment]
do not need you to explain the git principle, you just need to tell me what's going to do next?
Now I created the .git folder with the (git init) in the / Users / lee / test folder, and I created the .git file in the .git folder (git add.).
Now what should I do and give me the actual steps.
.git file should i edit? The The The The
NO.1 ...
NO.2 ...
NO.3 ...
I know that they need to be linked, but what exactly should I do?
I just learned programming, I do not understand git.
Also this is through Google Translate.

**Comment by deathaxe** [comment]
> do not need you to explain the git principle, you just need to tell me what's going to do next?

My last post contains the required steps. Nothing else needed.

> ... and I created the .git file in the .git folder (git add.).

???  not required! Too much!

**Comment by xxxlee** [comment]
I learned the knowledge of the next git, found the original so simple. I thought that your plugin was compared with the last saved version. Now I know and git library version to do comparison. I did not learn git before I was confused by your words. I thought it was the reason here.
![2017-10-15 11 22 22](https://user-images.githubusercontent.com/29938346/31581427-7a8260ea-b131-11e7-968e-a7793adcc570.png)
Did not learn git with this plugin, really will be confused.
Thank you very much!
